test/controllers/agent/RemoveClientDetailsSessionsControllerSpec.scala

```diff
@@ -16,77 +16,70 @@
 
 package controllers.agent
 
-import testConstants.BaseTestConstants.testAgentAuthRetrievalSuccess
-import config.featureswitch.FeatureSwitching
-import mocks.MockItvcErrorHandler
-import mocks.auth.MockFrontendAuthorisedFunctions
-import mocks.controllers.predicates.MockAuthenticationPredicate
-import mocks.views.agent.MockConfirmClient
+import mocks.auth.MockAuthActions
+import mocks.views.agent.MockEnterClientsUTR
+import play.api.Application
 import play.api.http.Status
-import play.api.mvc.MessagesControllerComponents
+import play.api
 import play.api.test.Helpers._
-import testUtils.TestSupport
 import uk.gov.hmrc.auth.core.BearerTokenExpired
+import views.html.agent.EnterClientsUTR
 
-class RemoveClientDetailsSessionsControllerSpec extends TestSupport
-  with MockConfirmClient
-  with MockAuthenticationPredicate
-  with MockFrontendAuthorisedFunctions
-  with FeatureSwitching
-  with MockItvcErrorHandler {
-
-  object TestRemoveClientDetailsSessionsController extends RemoveClientDetailsSessionsController(
-    mockAuthService,
-    testAuthenticator
-  )(
-    app.injector.instanceOf[MessagesControllerComponents],
-    appConfig,
-    mockItvcErrorHandler,
-    ec
-  )
-
-  ".show" when {
-    "the user is not authenticated" should {
-      "redirect the user to authenticate" in {
-        setupMockAgentAuthorisationException()
-
-        val result = TestRemoveClientDetailsSessionsController.show()(fakeRequestConfirmedClient())
-
-        status(result) shouldBe Status.SEE_OTHER
-        redirectLocation(result) shouldBe Some(controllers.routes.SignInController.signIn.url)
-      }
-    }
+class RemoveClientDetailsSessionsControllerSpec extends MockAuthActions
+  with MockEnterClientsUTR {
 
-    "the user has timed out" should {
-      "redirect to the session timeout page" in {
-        setupMockAgentAuthorisationException(exception = BearerTokenExpired())
+  override def fakeApplication(): Application = applicationBuilderWithAuthBindings()
+    .overrides(
+      api.inject.bind[EnterClientsUTR].toInstance(enterClientsUTR)
+    ).build()
 
-        val result = TestRemoveClientDetailsSessionsController.show()(fakeRequestConfirmedClient())
+  val testRemoveClientDetailsSessionsController = fakeApplication().injector.instanceOf[RemoveClientDetailsSessionsController]
 
-        status(result) shouldBe SEE_OTHER
-        redirectLocation(result) shouldBe Some(controllers.timeout.routes.SessionTimeoutController.timeout.url)
-      }
-    }
+  val fakeRequest = fakePostRequestConfirmedClient()
+
+    ".show" when {
```

Comment:
These tests are only testing for primary agents. You need to either wrap the tests in a loop (see ConfirmClientUTR which has done this)